contracts/programs/gumball-machine/src/lib.rs

```diff
@@ -318,12 +318,29 @@ pub mod gumball_machine {
         extension_len: u64,
         max_mint_size: u64,
         max_items: u64,
+        creator_keys: Vec<Pubkey>,
+        creator_shares: Vec<u8>
     ) -> Result<()> {
         let mut gumball_machine_data = ctx.accounts.gumball_machine.try_borrow_mut_data()?;
         let (mut header_bytes, config_data) =
             gumball_machine_data.split_at_mut(std::mem::size_of::<GumballMachineHeader>());
         let gumball_header = GumballMachineHeader::load_mut_bytes(&mut header_bytes)?;
         let size = max_items as usize;
+
+        // Construct creators array
+        let mut creators: [Creator; NUM_CREATORS] = [Default::default(), Default::default(), Default::default(), Default::default(), Default::default()];
+        assert_eq!(creator_keys.len(), creator_shares.len());
+        assert!(creator_keys.len() < NUM_CREATORS, "Cannot set more than {} creators", NUM_CREATORS);
+        assert!(creator_shares.iter().sum::<u8>() <= 100, "Cannot have creator royalty percentages total more than 100% of the sale price");
```

Comment:
I think adding a check for ` ... == 100` is useful, because we also don't want over 100 percent either